### PR TITLE
Update auth email sender and change notifications email to SendGrid

### PR DESCRIPTION
**Pull request by itstauq**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: syftbox/server/emails/router.py [hunk 1]
```diff
@@ -17,21 +17,20 @@ async def send_email(
     email_request: SendEmailRequest,
     server_settings: ServerSettings = Depends(get_server_settings),
 ) -> bool:
-    if not server_settings.email_service_api_key:
+    if not server_settings.sendgrid_secret:
         raise httpx.HTTPStatusError("Email service API key is not set", request=None, response=None)
```
Comment (eelcovdw) [inline syftbox/server/emails/router.py:21]: Suggestion: we can use `fastapi.HTTPException` for any endpoint exceptions, anything else probably gets transformed into a 50x by starlette

File: syftbox/server/emails/constants.py [hunk 2]
```diff
@@ -1,2 +1,2 @@
-EMAIL_SERVICE_API_URL = "https://api.resend.com/emails"
+EMAIL_SERVICE_API_URL = "https://api.sendgrid.com/v3/mail/send"
 FROM_EMAIl = "SyftBox <notifications@syftbox.openmined.org>"
```
Comment (eelcovdw) [inline syftbox/server/emails/constants.py:2]: nit: typo, lowercase l

Comment (eelcovdw) [inline syftbox/server/emails/constants.py:2]: nit: lowercase L typo

File: syftbox/server/emails/constants.py [hunk 3]
```diff
@@ -1,2 +1,2 @@
-EMAIL_SERVICE_API_URL = "https://api.resend.com/emails"
+EMAIL_SERVICE_API_URL = "https://api.sendgrid.com/v3/mail/send"
```
[the remark this answers is not in the document]
Comment (itstauq) [reply syftbox/server/emails/constants.py:1]: Moved it to `syftbox.lib.constants`

**Review by eelcovdw — APPROVED** [review]
LGTM!

**Pull request merged**